File: backend/app/api/health.py
"""
Health check endpoints
"""
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from ..core.database import get_db
from datetime import datetime

router = APIRouter()


@router.get("/health")
async def health_check(db: Session = Depends(get_db)):
    """Check if the service is healthy"""
    try:
        # Check database connection
        db.execute("SELECT 1")
        
        return {
            "status": "healthy",
            "timestamp": datetime.utcnow().isoformat(),
            "services": {
                "database": "connected",
                "api": "running"
            }
        }
    except Exception as e:
        return {
            "status": "unhealthy",
            "timestamp": datetime.utcnow().isoformat(),
            "error": str(e)
        }


@router.get("/ready")
async def readiness_check():
    """Check if the service is ready to accept traffic"""
    return {
        "status": "ready",
        "timestamp": datetime.utcnow().isoformat()
    }

File: m-eye/crates/api/src/api/health.rs
//! Health check endpoints

use axum::{Json, Router, extract::State, http::StatusCode};
use serde_json::json;
use meye_db::DatabasePool;

pub fn routes() -> Router<DatabasePool> {
    Router::new()
        .route("/", axum::routing::get(health_check))
        .route("/ready", axum::routing::get(readiness_check))
}

/// Basic health check
async fn health_check() -> Json<serde_json::Value> {
    Json(json!({
        "status": "healthy",
        "service": "m-eye-api"
    }))
}

/// Readiness check (database connection)
async fn readiness_check(State(pool): State<DatabasePool>) -> Result<Json<serde_json::Value>, StatusCode> {
    if pool.health_check().await {
        Ok(Json(json!({
            "status": "ready",
            "database": "connected",
            "redis": "connected"
        })))
    } else {
        Err(StatusCode::SERVICE_UNAVAILABLE)
    }
}
